site/ruby_scripts/tool.rb

```diff
@@ -0,0 +1,107 @@
+# encoding: utf-8
+
+
+module JekyllPlanet
+
+  def self.run( args )
+    Tool.new.run( args )
+  end
+
+  def self.main
+    self.run( ARGV )
+  end
+
+class Tool
+
+  def initialize()
+    puts "db settings:"
+    @db_config = {
+      adapter: 'sqlite3',
+      database: './planet.db'
+    }
+
+    pp @db_config
+  end
+
+
+  def run( args )
+    unless File.exists?( @db_config[:database])
+      puts "** error: database #{@db_config[:database]} missing; please check pluto documention for importing feeds etc."
+      exit 1;
+    end
+
+    Pluto.connect( @db_config )
+
+    Pluto::Model::Item.latest.limit(10).each_with_index do |item,i|
+      puts "[#{i+1}] #{item.title}"
+  
+      generate_blog_post( item )
+    end
+  end
+
+
+  def generate_blog_post( item )
+
+    posts_root = "./_posts"
+
+    FileUtils.mkdir_p( posts_root )  ## make sure path exists
+
+    ## Note:
+    ## Jekyll pattern for blogs must follow
+    ##  2014-12-21-  e.g. must include trailing dash (-)
+    fn = "#{posts_root}/#{item.published.strftime('%Y-%m-%d')}-#{title_to_key(item.title)}.html"
+
+    frontmatter =<<EOS
+---
+title:      "#{item.title}"
```

Comment:
Is this using string interpolation for everything except the content right now? 

I think that's what might be causing all the escaping issues.

An alternative is to load it into a ruby object and serialize to yaml. I found [one exmaple](https://stackoverflow.com/questions/17576472/convert-ruby-hash-into-yaml) of how to do that. I think that'd be much more cleaner

Reply:
You can also use `|` and `>` in YAML if you escape newline chars. ctrl+f for them on https://learnxinyminutes.com/docs/yaml/ for a brief explanation